content/blog/2024-08-01-clearlydefined.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Mixing license information into your GUAC
+date: 2024-08-01
+authors: ["Ben Cotton"]
+include_footer: true
+---
+
+[GUAC v0.8.0](../2024-07-25-guac-v0.8.0), which was released last week, includes the addition of a new data source for GUAC users: license data from [ClearlyDefined](https://clearlydefined.io/about).
+While GUAC previously supported reading license information from software bills of materials (SBOMs), adding support for ClearlyDefined represents a big step forward.
+ClearlyDefined provides a vetted and accurate representation of license information for a given release of software.
+
+Software licenses aren’t strictly a security concern, but they’re a key fact about your dependencies.
+Full software supply chain observability includes knowing the licenses in your supply chain, which helps you ensure you’re following license terms and your organizational requirements.
+
+## How GUAC uses ClearlyDefined
+
+GUAC maintainer Parth Patel decided to implement ClearlyDefined support as a certifier instead of a collector in order to ensure GUAC re-runs the query.
+GUAC certifiers run on a scheduled basis to capture up-to-date information that may have changed since the last run.
+License data is always incomplete because developers are always shipping new software releases, so this regular query ensures GUAC captures new license information.
+If you choose, you can also have GUAC query ClearlyDefined when you ingest a new SBOM, although this does slow down the ingestion process.
+
+GUAC is a tool for giving the information you need to make decisions, not to make decisions for you.
+As a result, GUAC does not try to guess which response is accurate if the information in the SBOM and the information from ClearlyDefined conflict.
+Both result in the creation of a CertifyLegal node in the graph, so you can decide which is more trustworthy on a case-by-case basis.
+
+Internally, GUAC uses [package URLs](https://github.com/package-url/purl-spec) (pURLs) to identify specific software package releases.
+However, ClearlyDefined uses a scheme they call “coordinates”.
+Part of implementing support in GUAC meant developing a library to convert pURLS to coordinates.
+Working with the ClearlyDefined community, Parth was able to develop a reference for mapping between the two systems.
+In a great example of cross-community collaboration, ClearlyDefined added [comprehensive documentation for coordinates](https://docs.clearlydefined.io/docs/resources/coordinates#purl-type-mapping) based on Parth’s work.
+
+One note of caution: ClearlyDefined’s API does not currently support batched queries. As a result, processing large dependency graphs may take longer due to rate limiting. Issue [#1168](https://github.com/clearlydefined/service/issues/1168) is open to add batched queries.
```

Comment:
I wonder if a bit more color on this paragraph can be added:
1. is this issue something we can expect to be fixed?
2. are patches welcome? can this be something that the cd community could benefit from a guac community member helping with?

Reply:
I don't think we can say at this point.